build.sh

```diff
@@ -745,7 +770,9 @@ fi
 
 daphne_msg "Build Daphne"
 
-cmake -S "$projectRoot" -B "$daphneBuildDir" -G Ninja $BUILD_CUDA $BUILD_ARROW $BUILD_FPGAOPENCL $BUILD_DEBUG \
+cmake -S "$projectRoot" -B "$daphneBuildDir" -G Ninja $BUILD_CUDA $BUILD_ARROW $BUILD_FPGAOPENCL  $BUILD_DEBUG \
+  -DMPI_INST_DIR=$MPIInstDirName \
+  -DCMAKE_C_COMPILER=mpicc -DCMAKE_CXX_COMPILER=mpic++ \
```

Comment:
I did not try to compile this but it looks like trouble changing to a custom compiler here. I suggest to separately compile the relevant sources into something like a libMPIKernels.so with another cmake --build call where you change the compiler setup.